worlds/AutoWorld.py

```diff
@@ -538,6 +538,10 @@ def remove(self, state: "CollectionState", item: "Item") -> bool:
             return True
         return False
 
+    def reached_region(self, state: "CollectionState", region: "Region") -> None:
+        """Called when a region is newly reachable by the state."""
+        pass
+
```

Comment:
I think this is an okay addition, but I would like to see some benchmarking done on it - I.e., how much does this slow down generation for worlds that don't use it

Reply:
Nice! That's very encouraging, considering how useful this feature will be.